tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -1202,10 +1250,20 @@ bool IsMatchedMatMulBiasAddAndGeluExact(
   // Find GeluExact
   std::map<string, int> dummy_matched_nodes_map;
   std::set<int> dummy_remove_node_indices;
-  return graph_matcher.GetMatchedNodes(
+  bool found_gelu_exact = graph_matcher.GetMatchedNodes(
       *gelu_exact_pattern, ctx.nodes_to_preserve, node_view,
       matched_nodes_map ? matched_nodes_map : &dummy_matched_nodes_map,
       remove_node_indices ? remove_node_indices : &dummy_remove_node_indices);
+  if (found_gelu_exact) {
+    return true;
+  } else {
+    matched_nodes_map->clear();
+    remove_node_indices->clear();
+  return graph_matcher.GetMatchedNodes(
+        *gelu_exact_pattern2, ctx.nodes_to_preserve, node_view,
+      matched_nodes_map ? matched_nodes_map : &dummy_matched_nodes_map,
+      remove_node_indices ? remove_node_indices : &dummy_remove_node_indices);
+  }
```

Comment:
Nit: Let's make this more concise:
```suggestion
  if (!matched_nodes_map) matched_nodes_map = &dummy_matched_nodes_map;
  if (!remove_node_indices) remove_node_indices = &dummy_remove_node_indices;
  if (graph_matcher.GetMatchedNodes(*gelu_exact_pattern, ctx.nodes_to_preserve,
                                    node_view, matched_nodes_map,
                                    remove_node_indices)) {
    return true;
  }
  matched_nodes_map->clear();
  remove_node_indices->clear();
  return graph_matcher.GetMatchedNodes(*gelu_exact_pattern2,
                                       ctx.nodes_to_preserve, node_view,
                                       matched_nodes_map, remove_node_indices);
```